src/main/java/com/entropyteam/entropay/employees/clients/services/ClientService.java

```diff
@@ -0,0 +1,88 @@
+import com.entropyteam.entropay.employees.common.exceptions.ResourceNotFoundException;
+import com.entropyteam.entropay.employees.common.mappers.ClientMapper;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class ClientService {
+
+    private final ClientRepository clientRepository;
+
+    public ClientSaveResponseDto createClient(ClientSaveRequestDto newClientDto) {
+        Client newClient = ClientMapper.MAPPER.toEntity(newClientDto);
+        newClient.setActive(true);
+
+        log.info("Attempting to save client {}.", newClientDto.getName());
+
+        ClientSaveResponseDto result = ClientMapper.MAPPER.toSaveResponseDto(clientRepository.save(newClient));
+
+        log.info("Successfully created client with name: {}.: Id: {}.", result.getName(), result.getId());
+
+        return result;
+    }
+
+    public ClientDto findClientById(String clientId) {
+        log.debug("Attempting to find client with Id: {}.", clientId);
+        return ClientMapper.MAPPER.toDto(clientRepository.findByIdAndIsActive(UUID.fromString(clientId), true)
+                .orElseThrow(() -> new ResourceNotFoundException("Client does not exist: " + clientId)));
+    }
+
+    public ClientSaveResponseDto updateClient(String clientId, ClientSaveRequestDto clientDto) {
+        Client client = clientRepository.findByIdAndIsActive(UUID.fromString(clientId), true)
+                .orElseThrow(() -> new ResourceNotFoundException("Client does not exist: " + clientId));
+
+        log.info("Attempting to Update client with Id: {}.", clientId);
+        log.trace("Current Client Data: {}.", client);
+
+        client.setAddress(clientDto.getAddress());
+        client.setContact(clientDto.getContact());
+        client.setPreferredCurrency(clientDto.getPreferredCurrency());
+
+        log.info("Client Data to update: {}.", client);
+
+        client = clientRepository.save(client);
+
+        log.info("Successfully updated client Id: {}.", clientId);
+
+        return ClientMapper.MAPPER.toSaveResponseDto(client);
+    }
+
+    public void deleteClient(String clientId) {
+        Client client = clientRepository.findById(UUID.fromString(clientId))
+                .orElseThrow(() -> new ResourceNotFoundException("Client does not exist: " + clientId));
+
+        log.debug("Attempting to delete client with Id: {}.", clientId);
+        log.trace("Current client Data: {}.", client);
+
+        client.setActive(false);
+        client.setModifiedOn(LocalDateTime.now());
+        clientRepository.save(client);
+
+        log.info("Successfully deleted client with Id: {}.", clientId);
+    }
+
+    public List<ClientDto> listActiveClients() {
+        log.debug("Attempting to retrieve the list of active clients.");
+        List<ClientDto> result = ClientMapper.MAPPER.toDtos(clientRepository.findAllByIsActive(
+                true, Sort.by(Direction.ASC, "name")));
```

Comment:
shouldn't be a parameter the order? asc/desc and column?

Reply:
Ok, done.